src/stdlib_64_bit_spookyv2_hashes.fypp

```diff
@@ -0,0 +1,718 @@
+                                              0_int64, sc_numvars), h )
+                call spookyhash_mix(                                       &
+                    transfer(spooky % data(sc_blocksize:2*sc_blocksize-1), &
+                             0_int64, sc_numvars), h )
+                p8 = prefix
+                length = length - prefix
+            end block
+        else
+            p8 = 0
+        end if
+
+    ! handle all whole blocks of sc_blocksize bytes requiring aligned bytes
+        bend = p8 + 8*(length/sc_blocksize)*sc_numVars
+        remainder = length - ( bend - p8 )
+        do while( p8 < bend )
+            spooky % data(0:sc_blocksize-1) = key( p8:p8+sc_blocksize-1 )
+            call spookyhash_mix( transfer( spooky % data(0:sc_blocksize-1), &
+                                 0_int64, sc_numvars), h )
+            p8 = p8 + sc_blocksize
+        end do
+
+    ! stuff away the last few bytes
+        spooky % remainder = remainder
+
+        if ( remainder > 0 ) then
+            spooky % data(0:remainder-1) = &
+                key(bend:bend+remainder-1)
+        end if
+
+    ! stuff away the variables
+        spooky % state(0:11) = h(0:11)
+
+    end subroutine spookyhash_update
+
+
+! report the hash for the concatenation of all message fragments so far
+     module subroutine spookyhash_final(spooky, hash_code)
+        type(spooky_subhash), intent(inout) :: spooky
+        integer(int64), intent(inout)       :: hash_code(2)
+
+        integer(int64) :: h(0:11)
+        integer(int64) :: index, remainder
+        integer(int8)  :: dummy(2)
+
+    ! init the variables
+        if ( spooky % length < sc_buffsize ) then
+            hash_code = spooky % state(0:1)
+            call spookyhash_short( spooky % data(0:spooky % length-1), &
+                                   hash_code )
+            return
+        end if
+
+        remainder = spooky % remainder
+
+        h(0:11)  = spooky % state(0:11)
+
+        if ( remainder >= sc_blocksize ) then
+          ! m_data can contain two blocks; handle any whole first block
+            call spookyhash_mix( transfer( spooky % data, 0_int64, &
+                                           2*sc_numvars), h )
+            index = sc_blocksize
+            remainder = remainder - sc_blocksize
+        else
+            index = 0
+        end if
+
+    ! mix in the last partial block, and the length mod sc_blocksize
+        spooky % data(sc_blocksize+remainder:) = 0_int8
+        dummy = transfer( remainder, 0_int8, 2 )
+
+        if  ( little_endian ) then
+            spooky % data(sc_blocksize-1) = dummy(1)
+        else
+            spooky % data(sc_blocksize-1) = dummy(2)
+        end if
+
+    ! do some final mixing
+        call spookyhash_end( transfer(spooky % data, 0_int64, 2*sc_numvars), h )
+
+        hash_code(1:2) = h(0:1)
+
+    end subroutine spookyhash_final
+
+
+    function rot_64_32( a, k )
```

Comment:
could be `pure`

Reply:
Done here and in `sodlib_64_bit_hash_functions.fypp`.

Reply:
FWIW `spooky_hash_update` had the wrong intent for an argument, but might also probably be pure, and probably `spooky_hash_final`.

Reply:
>  Done here and in ...

Did you push your commits? Because your changes do not appear in Github.

Reply:
I got confused with the two "could be `pure`" comments. I replied to the second one but the one I actually fixed was the first one. I have now fixed the second one.